core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -206,6 +206,21 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
     finalizeTask = Option(task)
   }
 
+  // Set the threshold to 1 billion which represents approximately 1GB of memory
+  // allocated to map output statuses.
```

Comment:
is the assumption based on `CompressedMapStatus`? For shuffle partition number exceeds `spark.shuffle.minNumPartitionsToHighlyCompress`(default is 2000), `HighlyCompressedMapStatus` is used, is it correct then?

Reply:
Thanks for pointing this out. Yes you are correct that `HighlyCompressedMapStatus` is to be used in this case. My understanding is that there's a bitmap and a normal map for huge blocks. In the ticket's discussion we only count the bitmap and acknowledge that is a very optimistic estimate. Since this is a warning message I prefer to be more cautious.

Reply:
I agree with your decision, and I'm OK with the user-oriented warning message, my only concern is this comment, we should point out that the memory consumption threshold is based on the worst case.

Reply:
+1 for @pan3793 's comment.

Reply:
Agree, this is not 1GB in size - the actual size can be quite variable.

Reply:
@y-wei Can you update the comment to explicitly state that this is the bitmap size estimation and the actual size is variable and can potentially be much larger?

Reply:
In reality, it will be quite a bit smaller - not larger.

Reply:
Why would it be smaller?

Reply:
This is detailed in `HighlyCompressedMapStatus` - please reference the class

Reply:
Hi @mridulm I've updated the comments for our rationale which points out that the `the actual size of `HighlyCompressedMapStatus` can be much larger than 128MB.`